app/models/concerns/slug.rb

```diff
@@ -1,13 +1,12 @@
-module Slug
+  module Slug
   def self.included(clazz)
     def clazz.from_param(token)
       id = if match = token.match(/.+-(\d+)/)
         match[1]
       else
         token
       end
-
-      found = where("id = ? OR #{slugger} ILIKE ?", id.to_i, token.tr('-', '%')).first
+      found = where(id: id.to_i).or(where("#{table_name}.#{slugger} ILIKE ?", token.tr('-', '%'))).first
```

Comment:
This change was necessary because the way administrate does BelongsTo queries lead to ambiguous queries where a join would contain both 'name' from Material and 'name' from 'Event'. 

You can see this error when trying to display a Material on production in the administrate UI right now.

This is currently marked as a security issue (SQL injection) by Hakiri but I fail to see how this is worse than before and I don't see an actual risk for injection even looking hard. I have no idea to mark the warning as a false positive though.

Somehow I think that there should be an easier way to create the table_name.slugger thing, but I would have to dive into the Rails code for that and I don't have the time to do that right now.

Reply:
ja, das ist ok, sind ja keine params